lib/sqlalchemy/dialects/postgresql/ranges.py

```diff
@@ -289,6 +289,9 @@ def __eq__(self, other: Any) -> bool:  # type: ignore[override]  # noqa: E501
         bounds inclusivity, returning ``True`` if they are equal.
         """
 
+        if not isinstance(other, self.__class__):
```

Comment:
to better support possible subclasses of range, maybe we could use `Range` directly there instead of `self.__class__`?

Reply:
Yes, but I fail to see the difference... can you explain?

Reply:
If someone were to subclass Range, then compare it with a Range this check would fail. I mean something like

```py
class MyRange(Range):
  pass

print(MyRange() == Range())
```
should return False since it fails the isinstance check (it may still return True, but that would by python that also calls the inverted eq check )

Reply:
I see, thanks. I fixed that.